### PR TITLE
SE-2010 Implements the XBlock problem response report with a detailed assessment and responses report

### PR DESCRIPTION
**Pull request by pomegranited**
Implements the `generate_report_data` method, allowing a custom report to be downloaded when using the Instructor Dashboard "Download Problem Response CSV" button.

The downloaded report contains the full list of enrolled learner submissions, and the assessments made against these submissions.

**JIRA tickets**: [OSPR-4073](https://openedx.atlassian.net/browse/OSPR-4073)

**Depencencies**: https://github.com/edx/edx-platform/pull/23109

**Merge deadline**: None

**Testing instructions**:

See https://github.com/edx/edx-platform/pull/23109 for sandbox and testing instructions.

[Sample report](https://github.com/edx/edx-ora2/files/5818640/OpenCraft_ORA2_2020_1_student_state_from_block-v1_OpenCraft%2BORA2%2B2020_1%2Btype%40chapter%2Bblock%4096abd57ebe08457cbc545a21f66b3240_2021-01-15-0525.csv.txt)

**Reviewers**
- [x] @swalladge
- [ ] edX reviewer[s] TBD

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @pomegranited! I've created [OSPR-4073](https://openedx.atlassian.net/browse/OSPR-4073) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by natabene** [comment]
@pomegranited Thank you for your contribution. Please let me know once it is ready to be looked at.

**Comment by swalladge** [comment]
@pomegranited 

> 4. To generate the report, run from an LMS shell:

Does this step in the test instructions need to be updated? That report command is no longer.

**Comment by pomegranited** [comment]
Thanks for your review @swalladge !  I'm going to squash these changes now so we can more easily cherry-pick them to the client's test servers.  CC @clemente

**Comment by swalladge** [comment]
@pomegranited :+1: to latest changes, with question:

- are we wanting average or median scores in the report? Median is usually the middle score, but the report appears to be calculating the average.

- [x] I tested this
- [x] I read through the code
- [x] I checked for accessibility issues
- [x] Includes documentation

File: openassessment/data.py
Comment (pomegranited) [inline openassessment/data.py:14]: To do:

```suggestion
from urllib.parse import urljoin
```

File: openassessment/tests/test_data.py
Comment (pomegranited) [inline openassessment/tests/test_data.py:15]: To do: `range` and `zip` are built into python3, remove this import.

**Comment by natabene** [comment]
@pomegranited Is this ready for edX review?

**Comment by pomegranited** [comment]
Hi @natabene -- looks like I've got some conflicts to resolve, so will get it ready for review and the sandbox redeployed next week, then it will be ready for review again.

**Comment by pomegranited** [comment]
Code conflicts resolved, but now we're blocked on the celery autodiscovery issues which are currently being investigated by edX, cf https://github.com/edx/edx-platform/pull/25479#issuecomment-736073800

**Comment by pomegranited** [comment]
@natabene Celery issues seem to be resolved in latest master, so the sandbox is working now, and so this PR is unblocked, and ready for review again! Maybe @jansenk or @nsprenkle would like to review?

**Comment by pomegranited** [comment]
@jnlapierre -- would you be able to help find a reviewer for this OSPR? CC @natabene

**Comment by natabene** [comment]
@pomegranited Could you please resolve conflicts first?

**Comment by pomegranited** [comment]
@natabene  Conflicts resolved, thank you! CC @jnlapierre

**Comment by pomegranited** [comment]
FYI @natabene the conflicts have re-occurred, which is likely since ora2 master is in active development. But they're simple ones to resolve, just with the setup.py and the translations files.

@jnlapierre  Do you have an idea of when this can be reviewed?

**Review by jnlapierre — APPROVED** [review]
LGTM, please resolve conflicts in preparation for merge.

**Comment by jnlapierre** [comment]
@pomegranited , could you or your delegate possible resolved conflicts?  I will merge immediately afterwards.

**Comment by swalladge** [comment]
@jnlapierre CC @pomegranited I rebased on master, squashed commits, and fixed the conflicts.  Unfortunately there are some failing tests now that I can't work out; we'll wait for @pomegranited to get back to investigate with more context.

**Comment by pomegranited** [comment]
@jnlapierre  Sorry for the delay here! I've fixed the conflicts and tests are passing again.

Are you ok to squash and merge now, or do you want me to squash?

Thanks @swalladge  for keeping this moving!

**Comment by jnlapierre** [comment]
@pomegranited , our product manager had a question which I suspect you can quickly answer...

"Do we know if this report exists on our platform yet? Or is this a new button?"

Also, if you have a moment, could you fix the setup.py conflict?  If not, I will gladly do it.  I'm waiting on two fixes to hit production and will then focus on this PR.

Thank you very much for your contributions.

**Comment by pomegranited** [comment]
Hi @jnlapierre , thank you for your quick reply!

> "Do we know if this report exists on our platform yet? Or is this a new button?"

This report is generated by an existing button which outputs problem response data for any XBlock. This button was added to the platform by https://github.com/edx/edx-platform/pull/17824.

If you want to see it in action, check out the example csv linked from https://github.com/edx/edx-platform/pull/23109, or follow these instructions: 

1. Login as the `staff@example.com` / `edx` user on 
2. Visit [Open Response Assessment Test > Instructor Dashboard > Data Download](https://pr23109.sandbox.opencraft.hosting/courses/course-v1:OpenCraft+ORA2+2020_1/instructor#view-data_download)
3. Scroll down to the  "Select a section or problem", and select a section, subsection, or problem.
4. Click "Create a report of problem responses" and wait for the link to appear to download the CSV.

> Also, if you have a moment, could you fix the setup.py conflict? If not, I will gladly do it. I'm waiting on two fixes to hit production and will then focus on this PR.

I've fixed the conflict. Let me know if you need anything else done before this can be merged?

Thank you!

**Comment by pomegranited** [comment]
Hi @jnlapierre  I've resolved the translation conflicts (again), squashed, and bumped the version to `3.3.0`, since this is a relatively major behaviour change.

Are you still ok to merge? I can do it as core committer, but just want to make sure you still approve.

**Comment by edx-webhook** [comment]
@pomegranited 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-ora2&entry.867055334=https://github.com/edx/edx-ora2/pull/1350&entry.1484655318=https://github.com/pomegranited&entry.752974735=2020-02-01+10:18&entry.1917517419=2021-03-02+17:19&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by pomegranited** [comment]
Hooray!! Thank you for working with me to get this mrged, @jnlapierre !

**Pull request merged**
